src/Microsoft.Health.Fhir.Core/Models/ResourceChangeData.cs

```diff
@@ -0,0 +1,37 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Health.Fhir.Core.Models
+{
+    public class ResourceChangeData
+    {
+        public ResourceChangeData(long id, DateTime timestamp, string resourceId, short resourceTypeId, int resourceVersion, byte resourceChangeTypeId, string resourceTypeName)
```

Comment:
If anything shouldn't be null, use EnsureArg.IsNotNullOrEmpty

Reply:
Added null checks.